### PR TITLE
Integrate test-plan stage into feature-flow pipeline (v0.10.0)

### PR DESCRIPTION
**Pull request by kirich1409**
## Summary

Встраивает стадию **TestPlan + TestPlanReview** в `feature-flow` pipeline плагина `developer-workflow`. Pipeline теперь генерирует, ревьюит и сохраняет test-plan между plan-review и implement; `acceptance` потребляет его как primary source. Bugfix-flow не получает свою стадию — документируется ручной путь через `/generate-test-plan` + mount-as-existing. Один релиз v0.10.0 — bump всех 6 плагинов семейства.

Pipeline до:
```
research → [decompose] → [plan-review] → implement → acceptance → PR
```
Pipeline после:
```
research → [decompose] → [plan-review] → [TestPlan → TestPlanReview] → implement → acceptance → PR
                                                ↓ skip (trivial/refactor/internal/small)
                                                └───────────────────────────→ implement
```

Детальное исследование и декомпозиция — в `swarm-report/test-plan-integration-research.md` и `swarm-report/test-plan-integration-decomposition.md` (gitignored).

## Breaking Changes

Нет. Стадия default-on с автоматическим skip для trivial-изменений:
- single-file change без behavior change (AC delta = 0)
- pure refactor (нет новых AC)
- internal utility без внешнего контракта
- single-task decompose ≤2 S-задачи

Override: `--skip-test-plan` для force-off. Bug-профили уходят в `bugfix-flow` в Phase 0.2 до TestPlan gate — отдельное skip-условие не требуется.

Существующие `docs/testplans/*` файлы автоматически монтируются через Pre-check в feature-flow Phase 1.5 без регенерации.

## Release Notes — v0.10.0

**Новые возможности `developer-workflow`:**
- Receipt-based test-plan артефакты: `swarm-report/<slug>-test-plan.md` (receipt) + `docs/testplans/<slug>-test-plan.md` (permanent)
- Phase segmentation permanent-файла при наличии decomposition
- Lightweight TC-формат (Given/When/Then) для non-UI features
- Inline чек-лист ревью test-plan в `plan-review` (5 пунктов: AC coverage, unhappy paths, edge cases, non-functional, priority alignment)
- Primary-source потребление test-plan в `acceptance` с fallback на on-the-fly генерацию
- Mount-as-existing для pre-orchestration файлов (orchestrator owns the write; acceptance Branch 2 — standalone fallback)
- Re-generation policy: rollback → append `## Regression TC` (любая severity когда происходит fix); spec change → explicit `--regenerate-test-plan`
- FAIL review → 3-cycle revise loop → эскалация с 3 опциями
- Документирован ручной путь test-plan для bugfix-flow через mount-as-existing (Phase 3)

**Миграция:** пользователи с существующими `docs/testplans/*` файлами ничего не делают — feature-flow автоматически монтирует их при следующем запуске. Для явной перегенерации — `--regenerate-test-plan`.

**Unified version bump:** все 6 плагинов семейства подняты на 0.10.0 (поверх 0.9.1 из #89). Family semver ranges расширены с `^0.9.1` на `^0.10.0` (caret при 0.x привязан к minor).

## Self-reference Note

PR v0.10.0 вводит test-plan механизм, но сам для себя его не применяет — срабатывает skip-условие #3 (internal utility without external contract change): изменения в SKILL.md и docs не затрагивают public API/endpoints. Self-skip намеренный; в SKILL.md заметка не сохраняется (per-PR meta не место в долгоживущем prompt) — фиксируется только здесь.

## Rebased onto 0.9.1

Ветка rebased на свежий main (включая PR #89 "Fix invalid plugin.json component paths"). Наши ранее bumped manifests унаследовали clean structure без `../skills` / `../agents` полей — kotlin-validator в изначальном smoke test был прав, main это подтвердил. Force-with-lease push выполнен.

## Known Limitations (pre-existing, not introduced by this PR)

- `developer-workflow/docs/ORCHESTRATION.md` — устаревшие упоминания скиллов `implement-task`, `test-feature`, `exploratory-test`
- `developer-workflow/docs/WORKFLOW.md §8` — не перечислены `write-spec` и `bug-hunt` в skill map
- `developer-workflow-swift` README — не упоминает `swiftui-design-system.md`
- `extend` — отсутствует README

Follow-up PR для чистки documentation — отдельной задачей.

## Test Plan (smoke test)

Метод: text-inspection (runtime-запуск через Skill tool невозможен — плагин кэширован, правки в worktree не подгрузятся).

6 сценариев прогнаны (`swarm-report/test-plan-integration-smoke-test.md`):

| # | Сценарий | Verdict |
|---|----------|---------|
| 1 | Default-on activation | PASS |
| 2 | Skip detection (trivial) | PASS |
| 3 | FAIL revise loop (≤3 циклов) | PASS |
| 4 | Mount-as-existing | PASS (после fix Pre-check в Phase 1.5) |
| 5 | Override `--skip-test-plan` | PASS |
| 6 | bugfix-flow fallback | PASS |

Automated validation: `bash scripts/validate.sh` зелёный; `plugin-dev:plugin-validator` PASS на всех 6 плагинах.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugins/developer-workflow/skills/feature-flow/SKILL.md [hunk 1]
```diff
@@ -118,6 +127,124 @@ If `swarm-report/<slug>-plan.md` or `swarm-report/<slug>-decomposition.md` was p
...
+  Receipt: `review_verdict: PASS`, `status: Ready`.
+- **WARN** — items (a)–(c) satisfied, but (d) or (e) violated. **Does not block.**
+  Transition to Implement. Receipt: `review_verdict: WARN`, warnings list enumerating the
+  violated items — preserved for downstream review and acceptance context. No revise-loop.
+- **FAIL** — any of (a), (b), (c) violated. Run the revise-loop: **TestPlan ← TestPlanReview**
+  up to 3 cycles. Each cycle patches the permanent test-plan file, re-reviews with the
+  same agents, and appends to the receipt's `Verdict History`. After 3 failed cycles →
+  **escalate to the user** with three options: (a) accept WARN manually and proceed,
+  (b) revise the spec and restart the pipeline, (c) use `--skip-test-plan` to bypass the
+  stage for this run.
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/feature-flow/SKILL.md]: Fixed in 8050287 — wording now points to plan-review state file's Verdict History; receipt itself carries only the latest review_verdict.

File: plugins/developer-workflow/skills/plan-review/SKILL.md [hunk 2]
```diff
@@ -143,6 +149,24 @@ Extract from the plan:
...
+Before moving to Step 2, determine whether the input is a **test-plan** artifact. Two signals,
+combined with OR — any one is enough:
+
+1. **YAML frontmatter type field** — the document starts with frontmatter that declares
+   `type: test-plan` (the permanent artifact at `docs/testplans/<slug>-test-plan.md`) or
+   `type: test-plan-receipt` (the receipt at `swarm-report/<slug>-test-plan.md`).
+2. **Structural signature** — the document contains a `## Test Cases` section AND discrete
+   TC blocks identifiable by TC-ID (e.g. `### TC-1`, `### TC-042`) AND explicit priority
+   labels from the set `P0` / `P1` / `P2` / `P3`. All three structural markers must be
+   present to count this signal.
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/plan-review/SKILL.md]: Fixed in 8050287 — added type: test-plan frontmatter to permanent template in generate-test-plan; Signal #2 regex now accepts TC-ID at any heading level (^#{1,6}\s+TC-[\\w-]+).

File: plugins/developer-workflow/skills/generate-test-plan/SKILL.md [hunk 3]
```diff
@@ -31,6 +31,75 @@ docs/testplans/<feature-name>-test-plan.md
...
+**Source spec:** <path or description>
+**Review verdict:** <verdict>
+
+## Phase Coverage
+- Phase 1 — TCs covered: TC-1..TC-N
+- Phase 2 — TCs covered: TC-N+1..TC-M
+
+## Review Findings
+(populated by plan-review stage; WARN items listed here)
+```
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/generate-test-plan/SKILL.md]: Fixed in 8050287 — dropped Review Findings body section; review findings live in frontmatter (review_warnings / review_blockers arrays) as the single source of truth.

File: plugins/developer-workflow/docs/WORKFLOW.md [hunk 4]
```diff
@@ -39,6 +39,14 @@ IDEA / FEATURE REQUEST
 [plan-review] ---- PoLL review of the plan (optional)
   |                  Artifact: plan review verdict
   v
+[test-plan] ---- Generate test plan (default-on, skip on 5 conditions or --skip-test-plan)
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/docs/WORKFLOW.md]: Fixed in cc06b35 — replaced hardcoded count with a link to feature-flow §TestPlan Stage Skip Detection.

File: plugins/developer-workflow/docs/WORKFLOW.md [hunk 5]
```diff
@@ -232,18 +240,25 @@ least one web-sourced insight. Relying solely on the codebase and training data
...
+                               |              ^                                                             |
+                               '-- FAIL ------'                                                             |
+                                                                                  ^                         |
+                                                                                  '-- review feedback ------'
+                                                          ^
+                                                          '-- Regression TC (Acceptance → Test Plan)
 ```
 
+Test Plan + Test Plan Review are **default-on**. They are skipped when one of the five
+skip-detector conditions holds (see `feature-flow` SKILL §TestPlan Stage Skip Detection)
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/docs/WORKFLOW.md]: Fixed in cc06b35 — same treatment: anchor to feature-flow §TestPlan Stage Skip Detection instead of a numeric count.

File: plugins/developer-workflow/skills/feature-flow/SKILL.md [hunk 6]
```diff
@@ -118,6 +127,124 @@ If `swarm-report/<slug>-plan.md` or `swarm-report/<slug>-decomposition.md` was p
...
+
+**Pre-check — mount existing permanent test plan:** before invoking
+`generate-test-plan`, check whether a pre-orchestration test plan already exists. If
+`docs/testplans/<slug>-test-plan.md` exists AND `swarm-report/<slug>-test-plan.md`
+receipt does NOT exist — this is a user-authored plan. Do NOT regenerate. The
+orchestrator owns this write: emit a mount-receipt following the canonical format from
+`generate-test-plan/SKILL.md` §Receipt (field overrides: `status: Mounted`,
+`review_verdict: skipped`, `source_spec: existing (pre-orchestration)`). Skip both
+TestPlan and TestPlanReview; announce **Stage: PlanReview → Implement (test plan mounted
+from existing)**. To regenerate, the user must re-invoke with `--regenerate-test-plan`.
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/feature-flow/SKILL.md]: Fixed in 8050287 — announcement templated as "Stage: <current> → Implement (test plan mounted from existing)" with a note that current can be PlanReview, Decompose, or Research.

File: plugins/developer-workflow/skills/plan-review/SKILL.md [hunk 7]
```diff
@@ -112,13 +113,18 @@ Status: {discovering | reviewing | synthesizing | fixing | done}
 - [ ] {agent2} — pending
 
 ## Verdict History
-### Cycle 1: {PASS | CONDITIONAL | FAIL}
+### Cycle 1: {PASS | CONDITIONAL | FAIL | WARN}
 - Blockers: {list}
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/plan-review/SKILL.md:118]: Fixed in 8050287 — added "Verdict:WARN → Done (test-plan branch only)" to the Allowed Transitions state machine.

File: plugins/developer-workflow/skills/acceptance/SKILL.md [hunk 8]
```diff
@@ -39,7 +39,54 @@ The specification defines what "correct" looks like. Accept any combination of:
...
+#### Branch 1 — Receipt present (`test_plan_source: receipt`)
+
+**Condition:** `swarm-report/<slug>-test-plan.md` exists (produced by `generate-test-plan`
+when invoked from the orchestrator).
+
+**Actions:**
+1. Read the receipt's YAML frontmatter and load `permanent_path`.
+2. Read `review_verdict`:
+   - `PASS` — proceed.
+   - `WARN` — proceed; carry WARN findings forward into the verification report as context.
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/acceptance/SKILL.md:57]: Fixed in 8050287 — Branch 1 now handles review_verdict: skipped explicitly (proceed, treat as user-authored mounted plan).

File: plugins/maven-mcp/package.json [hunk 9]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@krozov/maven-central-mcp",
-  "version": "0.9.1",
+  "version": "0.10.0",
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/maven-mcp/package.json:3]: Fixed in 24a1e48 — regenerated plugins/maven-mcp/package-lock.json; top-level and packages[""] now report 0.10.0.

File: plugins/developer-workflow/skills/generate-test-plan/SKILL.md [hunk 10]
```diff
@@ -31,6 +31,76 @@ docs/testplans/<feature-name>-test-plan.md
 Create the `docs/testplans/` directory if it doesn't exist. Use kebab-case for the feature name.
 Examples: `user-authentication-test-plan.md`, `cart-checkout-test-plan.md`.
 
+### Receipt (when invoked from orchestrator with a slug)
+
+When this skill is invoked from the `feature-flow` orchestrator, a `slug` argument is passed
+explicitly. In that case, in addition to the permanent document above, produce a **receipt**
+at `swarm-report/<slug>-test-plan.md` that the orchestrator and downstream stages
+(`plan-review`, `acceptance`) can consume for receipt-based gating.
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/generate-test-plan/SKILL.md:56]: Fixed in 6195a46 — adopted slug-based filename as the single canonical path (`docs/testplans/<slug>-test-plan.md`) for both orchestrated and standalone invocations. Standalone callers without an explicit slug derive one from the feature name via kebab-case (lowercase, runs of spaces/punctuation → `-`, trim edge `-`). Output section and receipt contract now consistent; feature-flow Phase 1.5 and acceptance Branch 2 mount by exact slug match without renaming.

File: plugins/developer-workflow/skills/generate-test-plan/SKILL.md [hunk 11]
```diff
@@ -31,6 +31,76 @@ docs/testplans/<feature-name>-test-plan.md
...
+
+- Permanent file generated at `docs/testplans/<feature-name>-test-plan.md` as described above.
+- No `swarm-report/<slug>-test-plan.md` is written.
+- No `phase_coverage` or receipt metadata tracked elsewhere.
+
+The `slug` parameter is therefore **mandatory only when invoked from the `feature-flow`
+orchestrator**. Standalone usage continues to work unchanged — existing callers,
+documentation references, and QA workflows that treat `docs/testplans/*-test-plan.md` as
+the single artifact are not broken by this change.
+
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/generate-test-plan/SKILL.md]: Fixed in 6195a46 — standalone usage now uses the canonical slug-based filename. When no slug is provided, it is derived from the feature name using the stable kebab-case convention: lowercase the name, replace runs of spaces or punctuation with `-`, trim leading/trailing `-` (e.g., "User authentication" → `user-authentication-test-plan.md`). Existing `docs/testplans/*-test-plan.md` files authored before this convention remain human-readable; mount logic matches only on the exact slug path.

File: plugins/developer-workflow/skills/bugfix-flow/SKILL.md [hunk 12]
```diff
@@ -128,6 +128,12 @@ Wait for `swarm-report/<slug>-implement.md` + `swarm-report/<slug>-quality.md`.
 
 ## Phase 3: Acceptance
 
+Bugfix-flow has no formal TestPlan stage — reproduction steps in `debug.md` act as the
+implicit test case. For bugs in critical flows that need a formal structured plan
+(regression protection, external QA handoff), run `/generate-test-plan` manually **before**
+`/bugfix-flow`: `acceptance` will detect the permanent file via mount-as-existing and use
+it as the primary source (see `acceptance/SKILL.md` §1.2 Branch 2).
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/bugfix-flow/SKILL.md]: Fixed in 6195a46 — bugfix-flow Phase 3 now instructs users to run `/generate-test-plan` with the **same slug** as the bugfix so the file lands at `docs/testplans/<slug>-test-plan.md`, or rename to match that exact path before `/bugfix-flow`. acceptance Branch 2 mounts by exact slug match.

File: plugins/developer-workflow/skills/acceptance/SKILL.md [hunk 13]
```diff
@@ -39,7 +39,58 @@ The specification defines what "correct" looks like. Accept any combination of:
...
+Phase 1.5 Pre-check. This branch runs only when `acceptance` is invoked outside of
+`feature-flow` (standalone QA session, `bugfix-flow`, user-triggered mid-flow), so the
+receipt has not been produced yet.
+
+**Actions:**
+
+1. Emit a mount-receipt at `swarm-report/<slug>-test-plan.md` following the canonical
+   format in `plugins/developer-workflow/skills/generate-test-plan/SKILL.md` §Receipt
+   with the mount overrides: `status: Mounted`, `review_verdict: skipped`,
+   `source_spec: existing (pre-orchestration)`, `phase_coverage: []`.
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/acceptance/SKILL.md]: Fixed in 766c9c8 — Branch 2 now derives `phase_coverage` from the permanent file by scanning for `### Phase N …` headings under `## Test Cases`. Empty list is used only when the file genuinely has no phase segmentation; when coverage cannot be determined reliably (malformed or mixed heading conventions), the field is omitted entirely rather than recording an incorrect empty list.

File: plugins/developer-workflow/skills/generate-test-plan/SKILL.md [hunk 14]
```diff
@@ -25,11 +25,101 @@ for a human QA engineer or the `manual-tester` agent to pick up later.
...
+   `slug` argument is passed explicitly. Use it as-is.
+2. **Standalone invocation, slug provided inline** — the user or caller may supply
+   a slug directly (e.g. `"slug: login-flow"`). Use it as-is.
+3. **Standalone invocation, no slug** — derive one from the feature name with the
+   stable kebab-case convention used elsewhere in workflow docs: lowercase the
+   name, replace runs of spaces or punctuation with `-`, trim leading/trailing `-`.
+
+Examples of derivation (rule 3): `"User authentication"` → `user-authentication-test-plan.md`,
+`"Cart & checkout"` → `cart-checkout-test-plan.md`, `"Token refresh (auth)"` →
+`token-refresh-auth-test-plan.md`.
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/generate-test-plan/SKILL.md]: Fixed in bf7c630 — rule 3 examples now show the derived slug (`user-authentication`, `cart-checkout`, `token-refresh-auth`); resulting filename shown separately as `docs/testplans/<slug>-test-plan.md`.

File: plugins/developer-workflow/skills/acceptance/SKILL.md [hunk 15]
```diff
@@ -185,6 +253,7 @@ the PR stage. `create-pr` references it for the PR description.
 **Type:** Feature / Bug fix
 **Spec source:** [what was used — requirements, debug.md reproduction steps, etc.]
 **Test plan:** [user-provided / generated from spec]
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/acceptance/SKILL.md]: Fixed in bf7c630 — Test plan field now reads "[resolved permanent path if sourced via receipt or mounted receipt | generated on-the-fly from spec | none]", matching the four-branch source selection in Step 1.2.

File: CLAUDE.md [hunk 16]
```diff
@@ -10,7 +10,7 @@ Monorepo for Claude Code plugins by krozov. Contains six plugins:
 |--------|-----------|-------------|
 | maven-mcp | `plugins/maven-mcp/` | MCP server for Maven dependency intelligence |
 | sensitive-guard | `plugins/sensitive-guard/` | Scans files for secrets and PII before they reach AI servers |
-| developer-workflow | `plugins/developer-workflow/` | Lifecycle pipeline — research, decomposition, spec, plan review, implementation, debugging, QA, PR workflow |
+| developer-workflow | `plugins/developer-workflow/` | Lifecycle pipeline — research, decomposition, spec, plan review, test planning, implementation, debugging, QA, PR workflow |
 | developer-workflow-experts | `plugins/developer-workflow-experts/` | 9 reusable review/consult agents (code-reviewer, architecture-expert, security-expert, …) — safe standalone |
 | developer-workflow-kotlin | `plugins/developer-workflow-kotlin/` | Kotlin/Android/KMP specialists and migration skills |
 | developer-workflow-swift | `plugins/developer-workflow-swift/` | Swift/iOS/macOS specialists and Swift/SwiftUI references |
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply CLAUDE.md:16]: Fixed — PR description updated to say 6 plugins. The "7 plugins" text was a leftover from before this branch was rebased onto main after PR #90 (Remove extend plugin); the actual version bumps in `marketplace.json` and all six plugin.json files already target the correct 6 plugins, so no code change was needed.
